tests/integration/query.ts

```diff
@@ -50,3 +58,121 @@ it('universal autocomplete sets search intents', async () => {
   expect(statefulCore.state.query.searchIntents).toEqual(['NEAR_ME']);
 });
 
+
+describe('ensure correct results from latest request', () => {
+  jest.useFakeTimers();
+  const mockedCore: any = {
+    verticalAutocomplete: jest.fn(
+      async (request: VerticalAutocompleteRequest) => {
```

Comment:
I think it would be a little easier to read if we extract these jest.fn() instantiations to a single method, we could have it take in a typescript generic for the request type

Reply:
I extracted the autocomplete jest.fn instantiations but for the universal/vertical search functions, I think we should keep it as is since their output is different.

Reply:
ah yeah I missed that!